tck-flow/src/test/java/org/reactivestreams/tck/flow/RangeFlowPublisherTest.java

```diff
@@ -0,0 +1,178 @@
+/************************************************************************
+ * Licensed under Public Domain (CC0)                                    *
+ *                                                                       *
+ * To the extent possible under law, the person who associated CC0 with  *
+ * this code has waived all copyright and related or neighboring         *
+ * rights to this code.                                                  *
+ *                                                                       *
+ * You should have received a copy of the CC0 legalcode along with this  *
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.*
+ ************************************************************************/
+
+package org.reactivestreams.tck.flow;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Flow;
+import java.util.concurrent.atomic.*;
+
+import org.reactivestreams.tck.TestEnvironment;
+import org.testng.annotations.*;
+
+@Test
+public class RangeFlowPublisherTest extends FlowPublisherVerification<Integer> {
+
+    static final Map<Integer, StackTraceElement[]> stacks = new ConcurrentHashMap<Integer, StackTraceElement[]>();
+
+    static final Map<Integer, Boolean> states = new ConcurrentHashMap<Integer, Boolean>();
+
+    static final AtomicInteger id = new AtomicInteger();
+
+    @AfterClass
+    public static void afterClass() {
+        boolean fail = false;
+        StringBuilder b = new StringBuilder();
+        for (Map.Entry<Integer, Boolean> t : states.entrySet()) {
+            if (!t.getValue()) {
+                b.append("\r\n-------------------------------");
+                for (Object o : stacks.get(t.getKey())) {
+                    b.append("\r\nat ").append(o);
+                }
+                fail = true;
+            }
+        }
+        if (fail) {
+            throw new AssertionError("Cancellations were missing:" + b);
+        }
+    }
+
+    public RangeFlowPublisherTest() {
+        super(new TestEnvironment());
+    }
+
+    @Override
+    public Flow.Publisher<Integer> createFlowPublisher(long elements) {
+        // FIXME how to do source dependencies between test scopes in gradle?
```

Comment:
I think it's better to have a native Flow-based implementation rather than using the RS version because the bridge will otherwise skipped due to `toRS(toFlow(rs)) == rs` "optimizing away" the two conversions.

Otherwise, here is a way to use test output from other modules:
https://github.com/akarnokd/RxJava3-preview/blob/master/interop/build.gradle#L7

Reply:
Cool, I'll remove the FIXME then I guess